### PR TITLE
New command !mdocstring

### LINKED ISSUE #14: Add an option to translate scene code to docstings with -docsting
image
Would be possible with this script.
```python

import re
from textwrap import indent
inputstring= """
class EllipseExample(Scene):
    def construct(self):
        ellipse_1 = Ellipse(width=2.0, height=4.0, color=BLUE_B)
        self.add(ellipse_1)
"""
inputstring= indent(inputstring, '    ')
template="""
\"""
Examples
--------

.. manim:: CLASSNAMEHERE
    :save_last_frame:

    CODEHERE
\"""
"""

outputstring=template.replace("CODEHERE", inputstring)
#possible_scene_names = ["Scene", "GraphScene"]
first_word = "class "
scene_name = "Scene" # does not support e.g. GraphScene yet. Maybe with a for loop ?
classname=re.search(fr'{first_word}(.*?).({scene_name})', inputstring).group(1)
outputstring= outputstring.replace("CLASSNAMEHERE", classname)
print(outputstring)

"""

outputstring=template.replace("CODEHERE", inputstring)
first_word = "class "
scene_name = "Scene" # does not support e.g. GraphScene yet. Maybe with a for loop ? #possible_scene_names = ["Scene", "GraphScene"]
classname=re.search(fr'{first_word}(.*?).({scene_name})', inputstring).group(1)
outputstring= outputstring.replace("CLASSNAMEHERE", classname)
print(outputstring)
```

### PR DESCRIPTION
**Pull request by naveen521kk**
Add a !mdocstring command
this will help form docstring and also format the code with black
Also, added poetry configuration and formatted the code base with black
Fixes #14

### CODE REVIEW

**Comment by infiniteregrets** [comment]
Can you explain a bit more about the !mdocstring command and possibly share a screenshot of what the end result looks like

**Comment by kolibril13** [comment]
This incorporates https://github.com/ManimCommunity/DiscordManimator/issues/14 (@infiniteregrets : you can find a screenshot there).
Thanks a lot, naveen, for this pull request!

**Comment by naveen521kk** [comment]
@infiniteregrets something like

**Comment by infiniteregrets** [comment]
Oh, I think this can be helpful for you guys! And this can be merged after #11, except you will have to make a few changes to your code. Great work!

**Comment by naveen521kk** [comment]
I think this can be merged first rather than that one. Since https://github.com/ManimCommunity/DiscordManimator/pull/11 does major changes on the whole structure it would easy to make changes there and fix conflicts rather than changing this here.

**Comment by kolibril13** [comment]
I don't mind in which order we merge these prs.
But it would be nice if this could be merged soon, as we have an event soon, where this docs feature could be really helpful!

File: DiscordManimator.py [hunk 1]
```diff
@@ -46,74 +53,255 @@ def construct(self):
...
+            scene_name = "Scene|GraphScene|VectorScene|LinearTransformationScene|MovingCameraScene|ZoomedScene|ReconfigurableScene|SampleSpaceScene|ThreeDScene|SpecialThreeDScene"
+            classregex = re.compile(
+                rf"class\s+(?P<classname>[A-Za-z]*)\(({scene_name})\)"
+            )
+            res = classregex.search(script)
+            if res:
+                class_name = res.group("classname")
+            else:
+                return {
+                    "content": "Can't find a unique Class Name. Either pass it as an argument or use standart methods."
```
Comment (Darylgolden) [inline DiscordManimator.py:192]:
```suggestion
                    "content": "Can't find a unique Class Name. Either pass it as an argument or use standard methods."
```

**Review by Darylgolden — CHANGES_REQUESTED** [review]
Minor typo, otherwise LGTM

**Comment by naveen521kk** [comment]
@kolibril13 does this look good to you?

Also, @behackl I have added poetry configurations, so while deploying you can either use a venv from poetry or run `poetry export -o requirements.txt` and run `pip install -r requirements.txt` while deploying.

**Comment by kolibril13** [comment]
> @kolibril13 does this look good to you?
> 
> Also, @behackl I have added poetry configurations, so while deploying you can either use a venv from poetry or run `poetry export -o requirements.txt` and run `pip install -r requirements.txt` while deploying.

yes, it looks good to me!
Thanks a lot again for the efforts you are doing here, this is a huge step to make it easier to contribute examples to the docs! :)

**Review by kolibril13 — APPROVED** [review]

**Comment by naveen521kk** [comment]
Ok merging :rocket:

**Comment by behackl** [comment]
Two remarks:

- Given that there is no automatic deployment setup for this, I'd appreciate if you'd let me merge -- otherwise it might happen that updates won't be deployed. (I don't really mind, it's just easier to miss stuff on my end.)
- In general, I like the idea of having a command that helps formatting docstrings -- but honestly, I think the actual work of composing the docstring should be made by Manim, potentially in a `manim docs` subcommand (which we should look into anyways, to make it easier to render single documentation pages).

The changes here are fine obviously, no need to revert anything. I'd still prefer PRing the formatting utility to the library itself. Only having it in our bot seems weird to me.

**Comment by kolibril13** [comment]
Do we have to restart the bot to apply the changes?

**Comment by naveen521kk** [comment]
> I'd appreciate if you'd let me merge 

:+1:. Maybe if it is possible we can set up something like automatic deployment using GHA.

> In general, I like the idea of having a command that helps formatting docstrings -- but honestly, I think the actual work of composing the docstring should be made by Manim, potentially in a manim docs subcommand (which we should look into anyways, to make it easier to render single documentation pages).

I am wondering how it would be helpful for a general user. I think it should be somewhere in the `scripts` folder in manim but shouldn't be build into Manim itself. Also, rendering single documentation separately seems like a nice and useful idea.

**Pull request merged**